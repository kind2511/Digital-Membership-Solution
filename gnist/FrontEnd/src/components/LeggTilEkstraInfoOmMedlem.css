.legg-til-ekstra-info-om-medlem-container {
  background-color: #f8f8f8;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  max-height: 270px;
}

.legg-til-ekstra-info-om-medlem-container .section-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}

.legg-til-ekstra-info-om-medlem-search-input {
  width: 95%;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.legg-til-ekstra-info-om-medlem-container .legg-til-ekstra-info-om-medlem-search-status {
  color: #666;
  margin-bottom: 10px;
}

.legg-til-ekstra-info-om-medlem-section-content {
  background-color: #fff;
  padding: 20px;
  border-radius: 8px;
  max-height: 110px;
  overflow-y: auto;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12);
  width: 93.3%;
}

.legg-til-ekstra-info-om-medlem-search-result {
  padding: 10px;
  border-bottom: 1px solid #eee;
  cursor: pointer;
}

.legg-til-ekstra-info-om-medlem-search-result:hover {
  background-color: #f7f7f7;
}

.additional-info-modal {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 20px;
  background-color: #f8f8f8;
  border-radius: 8px;
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.2);
  width: 50%;
  max-width: 500px;
}

.additional-info-modal h3 {
  color: #0056b3;
  font-weight: 600;
}

.additional-info-modal-textarea {
  width: 100%;
  min-height: 100px;
  font-family: Arial, sans-serif;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}

.additional-info-modal button {
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  font-size: 16px;
  cursor: pointer;
  margin-right: 10px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.15);
}

.additional-info-modal .save {
  background-color: #007BFF;
  color: white;
}

.additional-info-modal .close {
  background-color: #d3d3d3;
  color: black;
}

@media (max-width: 600px) {
  .additional-info-modal {
    width: 80%;
  }
}

.legg-til-ekstra-info-success-message {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #28a745;
  color: white;
  text-align: center;
  padding: 10px;
  font-weight: bold;
  z-index: 1500;
}

@keyframes slideInFromBottom {
  0% {
    bottom: 0;
    opacity: 0;
  }
  100% {
    bottom: 20px;
    opacity: 1;
  }
}
