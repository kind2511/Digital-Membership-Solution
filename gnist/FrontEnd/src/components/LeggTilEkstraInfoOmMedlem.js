import React from 'react';
import './LeggTilEkstraInfoOmMedlem.css';

function LeggTilEkstraInfoOmMedlem() {
    return (
        <div className="legg-til-ekstra-info-om-medlem-container">
            <h2 className="section-title">Legg til ekstra info om medlem</h2>
            <div className="section-content"></div>
        </div>
    );
}

export default LeggTilEkstraInfoOmMedlem;
