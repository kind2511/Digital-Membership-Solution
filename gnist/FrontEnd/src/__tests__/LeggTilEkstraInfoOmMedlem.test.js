import React from 'react';
import renderer from 'react-test-renderer';
import LeggTilEkstraInfoOmMedlem from '../components/LeggTilEkstraInfoOmMedlem';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const membersMock = [
  {
    auth0ID: 'auth0|123456789',
    first_name: 'John',
    last_name: 'Doe'
  },
  {
    auth0ID: 'auth0|987654321',
    first_name: 'Jane',
    last_name: 'Doe'
  }
];

mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/search_member/?name=').reply(200, membersMock);
mock.onPut('http://127.0.0.1:8000/digital_medlemsordning/add_member_info/auth0|123456789/').reply(200);

test('LeggTilEkstraInfoOmMedlem component snapshot', () => {
  const component = renderer.create(<LeggTilEkstraInfoOmMedlem />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
