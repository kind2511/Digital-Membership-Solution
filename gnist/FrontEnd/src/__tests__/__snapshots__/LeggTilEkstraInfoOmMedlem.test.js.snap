// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeggTilEkstraInfoOmMedlem component snapshot 1`] = `
<div
  className="legg-til-ekstra-info-om-medlem-container"
>
  <h2
    className="section-title"
  >
    Legg til ekstra info om medlem (IKKE SENSITIVE)
  </h2>
  <input
    className="legg-til-ekstra-info-om-medlem-search-input"
    onChange={[Function]}
    placeholder="SÃ¸k etter medlem..."
    type="text"
    value=""
  />
  <div
    className="legg-til-ekstra-info-om-medlem-search-status"
  />
  <div
    className="legg-til-ekstra-info-om-medlem-section-content"
  />
</div>
`;
